### DISCUSSION TITLE
openwrt-21.02编译失败

### DISCUSSION THREAD

**Discussion by jh0072110** [immortalwrt/immortalwrt#896] [General]
issue太难提了，直接在这里说吧，logs节选如下：
/workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/i40e-2.22.18/src/kcompat_defs.h:39:10: fatal error: kcompat_generated_defs.h: No such file or directory
 #include "kcompat_generated_defs.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[5]: *** [scripts/Makefile.build:262: /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/i40e-2.22.18/src/i40e_main.o] Error 1
make[4]: *** [Makefile:1739: /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/i40e-2.22.18/src] Error 2
make[3]: *** [Makefile:50: /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/i40e-2.22.18/.built] Error 2
make[4]: Leaving directory '/workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.4.231'
make[3]: Leaving directory '/workdir/openwrt/package/kernel/i40e'
time: package/kernel/i40e/compile#1.14#0.35#1.48
    ERROR: package/kernel/i40e failed to build.
make[2]: *** [package/Makefile:116: package/kernel/i40e/compile] Error 1
make[2]: Leaving directory '/workdir/openwrt'
make[1]: *** [package/Makefile:110: /workdir/openwrt/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/workdir/openwrt'
make: *** [/workdir/openwrt/include/toplevel.mk:230: world] Error 2
Error: Process completed with exit code 2.
错误相关的部分，希望有所帮助，需要完整logs的话我明日开电脑后会更新在帖子里，感恩！

**Comment by jh0072110** [comment]
[11_Compile the firmware.txt](https://github.com/immortalwrt/immortalwrt/files/10894061/11_Compile.the.firmware.txt)
update full logs

**Comment by 1715173329** [comment]
fixed

**Reply by jh0072110** [reply]
比心

**Comment by jh0072110** [comment]
糟糕啦，又编译失败啦（甚至连新discussions都懒得开了的我……）
logs节选：
ERROR: module '/workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/ixgbe-5.18.11/src/ixgbe.ko' is missing.
make[3]: *** [Makefile:46: /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/ixgbe-5.18.11/.pkgdir/kmod-ixgbe-vendor.installed] Error 1
make[3]: Leaving directory '/workdir/openwrt/package/kernel/ixgbe'
time: package/kernel/ixgbe/compile#0.22#0.05#0.26
    ERROR: package/kernel/ixgbe failed to build.
make[2]: *** [package/Makefile:116: package/kernel/ixgbe/compile] Error 1
make[2]: Leaving directory '/workdir/openwrt'
make[1]: *** [package/Makefile:110: /workdir/openwrt/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/workdir/openwrt'
make: *** [/workdir/openwrt/include/toplevel.mk:230: world] Error 2
完整logs：
[11_Compile the firmware.txt](https://github.com/immortalwrt/immortalwrt/files/10987456/11_Compile.the.firmware.txt)
